lib/pushmeup/apns/http2_client.rb

```diff
@@ -30,71 +31,26 @@ def send_notification(device_token, bundle_identifier, message)
     end
 
     def send_notifications(notifications, bundle_identifier)
-      @queue = notifications
-      send_next_notification(bundle_identifier)
-    end
-
-    def send_next_notification(bundle_identifier)
-      with_connection do
-        if notification = @queue.pop
-          stream = @conn.new_stream
-          stream.on(:close) { send_next_notification(bundle_identifier) }
-          json = notification.to_json
-          headers = {
-            ':scheme' => 'https',
-            ':method' => 'POST',
-            ':path' => "/3/device/#{notification.device_token}",
-            'authorization' => "bearer #{jwt_token}",
-            'content-type' => 'application/json',
-            'apns-topic' => bundle_identifier,
-            'content-length' => json.bytesize.to_s # should be less than or equal to 4096 bytes
-          }
-          stream.headers(headers, end_stream: false)
-          stream.data(json)
-          while !@ssl.closed? && !@ssl.eof?
-            data = @ssl.read_nonblock(1024)
-            begin
-              @conn << data
-            rescue => e
-              close_socket_and_ssl
-              raise
-            end
-          end
-        else
-          close_socket_and_ssl
-        end
-      end
-    end
-
-    def with_connection
-      if @ssl.nil? || @ssl.closed?
-        open_connection
+      notifications.each do |notification|
+        path = "/3/device/#{notification.device_token}"
+        request = @client.prepare_request(:post, path, headers: headers(bundle_identifier), body: notification.to_json)
+        @client.call_async(request)
+        @client.join
```

Comment:
if we close it we are not able to reuse the connection, I want the caller to be responsible for closing the connection.